kong/plugins/opentelemetry/handler.lua

```diff
@@ -150,6 +150,17 @@ function OpenTelemetryHandler:access()
   propagation_set("preserve", header_type, translate_span_trace_id(root_span), "w3c")
 end
 
+function OpenTelemetryHandler:header_filter(conf)
+  if conf.http_response_header_for_traceid then
+    local trace_id = kong.ctx.plugin.trace_id
+    if not trace_id then
+      local root_span = ngx.ctx.KONG_SPANS and ngx.ctx.KONG_SPANS[1]
+      trace_id = root_span and root_span.trace_id
+    end
+    kong.response.add_header(conf.http_response_header_for_traceid, trace_id)
+  end
```

Comment:
```suggestion
  if not conf.http_response_header_for_traceid then
    return
  end
  
  local trace_id = kong.ctx.plugin.trace_id
  if not trace_id then
    local root_span = ngx.ctx.KONG_SPANS and ngx.ctx.KONG_SPANS[1]
    trace_id = root_span and root_span.trace_id
  end
  kong.response.add_header(conf.http_response_header_for_traceid, trace_id)
```

Reply:
I think we can not do this, because the header filter may have more capabilities in the future, if add `return` early, it may be hard to extension this function.

Reply:
Code should reduce nesting where possible by handling error cases/special conditions first and returning early or continuing the loop. Reduce the amount of code that is nested multiple levels.

Ref: https://testing.googleblog.com/2017/06/code-health-reduce-nesting-reduce.html

Reply:
I suggest doing this, and I think it's a good habit to get into. If you don't think it's necessary, it's OK.😉